grails-app/services/org/pih/warehouse/core/OrganizationService.groovy

```diff
@@ -67,6 +67,13 @@ class OrganizationService {
         return organization
     }
 
+    Organization saveOrganization(Organization organization) {
+        if (organization.validate() && !organization.hasErrors()) {
+            return organization.save()
+        }
+        throw new ValidationException("Invalid organization", organization.errors)
+    }
+
```

Comment:
Here's an example from Peter Ledbrook, a prominent Grails developer. This might be from Grails 2, but seems to be a decent way of handling the save in the service.

https://github.com/GrailsInAction/graina2/blob/fd1e5ac24fc2708c69b44392e8674796fdb22da4/ch11/hubbub/grails-app/services/com/grailsinaction/PostService.groovy#L18

Here's an example from our codebase if we don't want to return anything from the save.

https://github.com/openboxes/openboxes/blob/1e77b282da20c34b75ebec47b896e3528dae095e/grails-app/services/org/pih/warehouse/data/LocationDataService.groovy#L121

And another one

https://github.com/streamr-dev/core-api/blob/9db3ae26621f504daa060c88d6a50571d8d3310d/grails-app/services/com/streamr/core/service/EthereumUserService.groovy#L28

And spring security 

https://github.com/grails/grails-spring-security-ui/blob/e71f47abe9648314bc304c15d76b487980d6209f/plugin/grails-app/services/grails/plugin/springsecurity/ui/SpringSecurityUiService.groovy#L352

And one that basically acts as a thin wrapper around GORM in order to add transactions.

https://github.com/grails/grails-spring-security-ui/blob/e71f47abe9648314bc304c15d76b487980d6209f/plugin/grails-app/services/grails/plugin/springsecurity/ui/SpringSecurityUiService.groovy#L352

So I think maybe something like this

```
Domain saveDomain(Domain object) { 
    if (object.validate() && object.save()) {
        return object
    } else {
        throw new ValidationException("Invalid object", object.errors)
    }
}
```
or just something more straightforward if we want the client to perform error handling
```
Domain saveDomain(Domain object) { 
    return object.save()
}
```

Reply:
Note we'll probably move this service to @Service so perhaps this doesn't matter. But I just wanted to go down this path because I want a decent solution in case we choose not to go down the @Service route.

Reply:
thanks for the resources @jmiranda , will take a look

Reply:
I might be wrong but I feel like even this code
```groovy
Domain saveDomain(Domain object) { 
    if (object.validate() && object.save()) {
        return object
    } else {
        throw new ValidationException("Invalid object", object.errors)
    }
}
```

could be shortended to just
```groovy
Domain saveDomain(Domain object) { 
   object.save(failOnError: true)
   return object
}
```

citing the grails [documentation](https://docs.grails.org/latest/ref/Domain%20Classes/save.html):

> When set to true the save method with throw a grails.validation.ValidationException if [validation](https://docs.grails.org/latest/guide/validation.html) fails.

Reply:
@jmiranda just wanted to point out, that the `save()` that is created by data services, throws `ValidationException` by default, so it doesn't return `null` in case, something was not saved. We can't manipulate it.

> If any of the parameters don’t match up to a property on the domain class then a compilation error will occur.

> If a validation error occurs then a ValidationException will be thrown from the service.

so it feels like Grails recommends this approach if they make it as default and unchangeable for data services.

Reply:
@kchelstowski Ooh, interesting. Thanks for that piece of information. 

![image](https://github.com/openboxes/openboxes/assets/1162935/43f6353e-ce96-4421-a102-28325ebf1e6e)